### ISSUE TITLE
Support for PrometheusRule contained in larger YAML manifest

### ISSUE THREAD

**Issue by felipesere** [cloudflare/pint#746]
We use `helm` to produce a single, large `manifest.yaml` for each application that contains many YAML docs.
When running `pint` against it, my expectation is that it would find the right documents with `kind: PrometheusRule` and then apply the lints there. At the moment, it just silently fails.
If I just extract the PrometheusRules from that `manifest.yaml` and run them individually pint correctly finds issues.

Here is our pint config:
```hcl
parser {
  relaxed = ["(.*)", "yaml/parse"]
}
checks {
  disabled = ["promql/fragile"] # some upstream kubernetes-mixins use without and it's their problem
}
# truelayer invariants
rule {
  match {
    kind = "alerting"
  }
  # all alerts must have a standard severity
  label "severity" {
    required = true
    value    = "error|critical|warning|info"
  }
  # all alerts should have owners as a label
  label "owners" {
    required = true
  }
}
```

**Comment by prymitive** [comment]
Unless you can provide a sample file there's not much we can do here.

**Comment by patsevanton** [comment]
For Example:
https://github.com/prometheus-operator/kube-prometheus/blob/main/manifests/prometheusOperator-prometheusRule.yaml
https://github.com/prometheus-operator/kube-prometheus/blob/main/manifests/alertmanager-prometheusRule.yaml

**Comment by felipesere** [comment]
👋 Circling back to this. I have a small file that I easily reproduces the issue I face:

```yaml
---
apiVersion: networking.k8s.io/v1
kind: NetworkPolicy
metadata:
  name: deployment-notifier
  namespace: tools
spec:
  podSelector:
    matchLabels:
      app: deployment-notifier
  policyTypes:
  - Ingress
  - Egress
  egress:
  - to:
    - namespaceSelector:
        matchLabels:
          name: opentelemetry-operator-system
    ports:
    - port: 1
---
apiVersion: monitoring.coreos.com/v1
kind: PrometheusRule
metadata:
  name: deployment-notifier
  namespace: tools
spec:
  groups:
  - name: ./deployment-notifier.rules
    rules:
    - alert: DeploymentNotifierSwarmiaProviderErrors
      annotations:
        summary: Something something...
      expr: 1 > 0
      for: 10m
      labels:
        severity: not-that-critical
```

Above yaml is a snippet of what my helm chart produces. If I keep the order as is I get zero lint failures with the config from my previous comment.
If I change the order of the documents to have the PromRule first, I get the expected lint error.

**Comment by prymitive** [comment]
That's because current yaml parser doesn't correctly read all documents from a multi-document file.
If you have a file with:

```
---
some:
  yaml
---
some more:
  yaml
``` 

then only first `some:yaml` document is read, that's a bug and I hope to fix it one day but don't have much time for it at the moment.

**Comment by prymitive** [comment]
Fixed in #894
